stream_alert/alert_processor/outputs.py

```diff
@@ -553,12 +547,14 @@ def dispatch(self, **kwargs):
 
         slack_message = self._format_message(kwargs['rule_name'], kwargs['alert'])
 
-        resp = self._request_helper(creds['url'], slack_message)
+        resp = self._get_request(creds['url'], slack_message)
         success = self._check_http_response(resp)
 
-        if not success:
-            LOGGER.error('Encountered an error while sending to Slack: %s',
-                         resp.read())
+        if not success and resp:
+            response = resp.json()
+            LOGGER.error('Encountered an error while sending to Slack: %s\n%s',
```

Comment:
Same comment as before, having this handled in the base class' `_check_http_response` method.